### ISSUE TITLE
ERR: Unsupported compression type: 3 causing seg fault

### ISSUE THREAD

**Issue by scowan1995** [ggreer/the_silver_searcher#1436]
I tried to search the bash manual for the string "-z" which resulted in  the following error:
```
ERR: Unsupported compression type: 3
[1]    8200 segmentation fault (core dumped)  ag "-z" bash_manual.txt
```
I suppose it is to do with -z being an option for ag. It did work with:
```
ag -- "-z" bash_manual.txt
```
This is on Ubuntu 20 using zsh.

Maybe ag should not treat quoted strings as potential options.

**Comment by JaapJoris** [comment]
I don't think this is an issue with `ag` specifically, although of course it shouldn't segfault. Many Unix utilities have the same "problem". Try for instance creating a file named `-z` with the command `touch -z`.

There is no difference between passing `"-z"` or `-z` as the argument. The shell will remove the quotes before passing the argument to `ag`.

Your solution is correct, by using `--` to indicate there will be no further option arguments. This works for `touch` too: `touch -- -z`. 
Good luck trying to remove this file now ;-)
